projects/plugins/boost/app/assets/src/js/features/critical-css/status/status.tsx

```diff
@@ -1,112 +1,120 @@
-	progress: number;
-	showRegenerateButton: boolean;
-	isCloudCssAvailable?: boolean;
-	issues: Provider[];
-	successCount?: number;
-	generateText?: string;
-	generateMoreText?: string;
+	cssState: CriticalCssState;
+	isCloud?: boolean;
+	hasRetried: boolean;
+	retry: () => void;
+	highlightRegenerateButton?: boolean;
+	extraText?: string; // Optionally, provide a sentence to use after the main message to provide more context.
+	overrideText?: string; // Optionally, provide a custom message to display instead of the default.
 };
 
 const Status: React.FC< StatusTypes > = ( {
-	status,
-	updated,
-	progress,
-	showRegenerateButton = false,
-	isCloudCssAvailable = false,
-	issues,
-	successCount = 0,
-	generateText = '',
-	generateMoreText = '',
+	cssState,
+	isCloud = false,
+	hasRetried,
+	retry,
+	highlightRegenerateButton = false,
+	extraText,
+	overrideText,
 } ) => {
 	const regenerateAction = useRegenerateCriticalCssAction();
+	const successCount =
+		cssState.providers.filter( provider => provider.status === 'success' ).length || 0;
+	const issues = getCriticalCssIssues( cssState );
 
+	// If there has been a fatal error, show it.
+	if ( isFatalError( cssState ) ) {
+		return (
+			<ShowStopperError
+				supportLink={ ( isCloud && 'https://jetpack.com/contact-support/' ) || undefined }
+				cssState={ cssState }
+				retry={ retry }
+				showRetry={ ! hasRetried }
+			/>
+		);
+	}
+
+	// If my parent has provided override text, show it.
+	if ( overrideText ) {
+		return (
+			<div className="jb-critical-css__meta">
+				<div className="summary">{ overrideText }</div>
+			</div>
+		);
+	}
+
+	// Otherwise, show the status.
 	return (
 		<div className="jb-critical-css__meta">
 			<div className="summary">
-				{ successCount === 0 ? (
-					<div className="generating">{ generateText }</div>
-				) : (
-					<>
-						<div className="successes">
-							{ sprintf(
-								/* translators: %d is a number of CSS Files which were successfully generated */
-								_n( '%d file generated', '%d files generated', successCount, 'jetpack-boost' ),
-								successCount
-							) }
-							{ updated && (
-								<>
-									{ ' ' }
-									<TimeAgo time={ new Date( updated * 1000 ) } />
-									{ '.' }
-								</>
-							) }
-							{ ! isCloudCssAvailable && (
-								<>
-									{ ' ' }
-									{ __(
-										'Remember to regenerate each time you make changes that affect your HTML or CSS structure.',
-										'jetpack-boost'
-									) }
-								</>
-							) }
-							{ progress < 100 && (
-								<>
-									{ ' ' }
-									<span>{ generateMoreText }</span>
-								</>
-							) }
-						</div>
+				<div className="successes">
+					{ sprintf(
+						/* translators: %d is a number of CSS Files which were successfully generated */
+						_n( '%d file generated', '%d files generated', successCount, 'jetpack-boost' ),
+						successCount
+					) }
+
+					{ cssState.updated && (
+						<>
+							{ ' ' }
```

Comment:
What's the purpose of this?

Reply:
@haqadn that's a literal space that ensures a space is rendered because react would otherwise strip it